packages/runtime/container-runtime/src/gcConfig.ts

```diff
@@ -0,0 +1,152 @@
+import { getGCVersion, IGCMetadata } from "./summaryFormat";
+import { IGCRuntimeOptions } from "./containerRuntime";
+import { runSessionExpiryKey } from "./garbageCollection";
+
+/** All known test mode configurations */
+export interface IGCTestConfig {
+    /**
+     Sweep V0 is a test mode for using a very short timeout in order to enable testing of real sweep behavior.
+     It's similar to testMode but rather than deleting immediately, it exercises the Sweep timers and codepaths
+    */
+    SweepV0?: {
+        sessionExpiryTimeoutMs: number;
+    };
+}
+
+/** All known test mode names */
+export type GCTestMode = keyof IGCTestConfig;
+
+export const oneDayMs = 1 * 24 * 60 * 60 * 1000;
+
+const defaultCacheExpiryTimeoutMs = 2 * oneDayMs; // 2 days, matches the same variable in odsp-driver
+export const defaultInactiveTimeoutMs = 7 * oneDayMs; // 7 days
+export const defaultSessionExpiryDurationMs = 30 * oneDayMs; // 30 days
+const defaultBufferMs = oneDayMs; // 1 day
+
+export type GcContainerConfig =
+    // Common props regardless of sweepAllowed
+    {
+        prevSummaryGCVersion?: number;
+    }
+    & (
+        // Different sets of props depending on sweepAllowed
+        | {
+            sweepAllowed: false;
+            gcAllowed: boolean;
+        }
+        | {
+            sweepAllowed: true;
+            gcAllowed: true;
+
+            testMode: GCTestMode | undefined;
+            snapshotCacheExpiryMs: number;
+            sessionExpiryTimeoutMs: number;
+            sweepTimeoutBufferMs: number;
+        }
+    );
+
+//* TODO: Add logging of different config sources and outcome (either here or in GC.ts)
+
+export function configForExistingContainer(
+    metadata?: IGCMetadata,
+): GcContainerConfig {
+    // Crash if we see a test mode we don't recognize, since we don't know how it's implemented or should be used
+    const testMode = metadata?.gcTestMode;
+    if (testMode !== undefined && testMode !== "SweepV0") {
+        unreachableCase(testMode, `Cannot open container created under unknown GC Test Mode [${testMode}]`);
+    }
+
+    // Existing documents which did not have metadata blob or had GC disabled have version as 0. For all
+    // other existing documents, GC is enabled.
+    const prevSummaryGCVersion = getGCVersion(metadata);
+    const gcAllowed = prevSummaryGCVersion > 0;
+    const sessionExpiryTimeoutMs = metadata?.sessionExpiryTimeoutMs;
+    if (!gcAllowed
+        || metadata?.sweepEnabled !== true
+        || sessionExpiryTimeoutMs === undefined
+    ) {
+        return {
+            sweepAllowed: false,
+            gcAllowed,
+            prevSummaryGCVersion,
+        };
+    }
+
+    const sweepV0: boolean = testMode === "SweepV0";
+    const snapshotCacheExpiryMs = sweepV0 ? 0 : defaultCacheExpiryTimeoutMs; // Ignore snapshot expiry for SweepV0
+    const sweepTimeoutBufferMs = metadata.sweepTimeoutBufferMs ?? defaultBufferMs; // For SweepV0 we expect it to be set
```

Comment:
if this is not set, we should not run sweepV0

Reply:
Yeah, I'm on the fence between overly-asserting the relationship between different settings v. just write the code to do the right thing and log enough to diagnose if something weird happens.  Code elsewhere (I think not written yet!) will in fact always write sweepTimeoutBufferMs if we're in SweepV0 mode.

I can add an assert if you think that's a better approach.  I don't think it will work well to disable SweepV0 if this is missing though - Likely other timeouts will be small, and we should keep the file marked as test-mode to keep it from leaking out to production environment.